alephclient/crawldir.py

```diff
@@ -112,14 +132,24 @@ def ingest_upload(self, path: PathLike, parent_id: str, foreign_id: str) -> str:
         if parent_id is not None:
             metadata["parent_id"] = parent_id
         result = self.api.ingest_upload(
-            self.collection_id, path if isinstance(path, Path) else Path(path.path), metadata=metadata, index=self.index
+            self.collection_id,
+            path if isinstance(path, Path) else Path(path.path),  # type: ignore
```

Comment:
I don't think we need to check the type again. We are already expecting `path` to be an instance of `Path`.

Reply:
Without this check the test_ingest test will fail as the assertion:

assert call in self.api.ingest_upload.mock_calls

Which is why I've kept it in

Reply:
I think it's failing because of the `get_foreign_id` call I talked about in the comment above. Fixing that should fix the broken test without needing this `isinstance` check